templates/todo/projects/csharp-sql/.repo/bicep/infra/main.bicep

```diff
@@ -54,139 +55,312 @@ resource rg 'Microsoft.Resources/resourceGroups@2021-04-01' = {
-    appSettings: {
-      AZURE_SQL_CONNECTION_STRING_KEY: sqlServer.outputs.connectionStringKey
+    appInsightResourceId: appInsightResourceId
+    managedIdentities: {
+      systemAssigned: true
+    }
+    siteConfig: {
+      cors: {
+        allowedOrigins: [ 'https://portal.azure.com', 'https://ms.portal.azure.com' ,'https://${web.outputs.defaultHostname}' ]
+      }
+      linuxFxVersion: 'dotnetcore|8.0'
+      alwaysOn: true
+      appCommandLine: ''
+    }
+    appSettingsKeyValuePairs: {
+      AZURE_SQL_CONNECTION_STRING_KEY: connectionStringKey
+      AZURE_KEY_VAULT_ENDPOINT: keyVault.outputs.uri
+      SCM_DO_BUILD_DURING_DEPLOYMENT: 'False'
+      ENABLE_ORYX_BUILD: 'True'
     }
   }
 }
 
 // Give the API access to KeyVault
-module apiKeyVaultAccess '../../../../../../common/infra/bicep/core/security/keyvault-access.bicep' = {
-  name: 'api-keyvault-access'
+module accessKeyVault 'br/public:avm/res/key-vault/vault:0.3.5' = {
+  name: 'accesskeyvault'
   scope: rg
   params: {
-    keyVaultName: keyVault.outputs.name
-    principalId: api.outputs.SERVICE_API_IDENTITY_PRINCIPAL_ID
+    name: keyVault.outputs.name
+    enableRbacAuthorization: false
+    accessPolicies: [
+      {
+        objectId: principalId
+        permissions: {
+          secrets: [ 'get', 'list' ]
+        }
+      }
+      {
+        objectId: api.outputs.systemAssignedMIPrincipalId
+        permissions: {
+          secrets: [ 'get', 'list' ]
+        }
+      }
+    ]
+    secrets:{
+      secureList: [
+        {
+          name: 'sqlAdmin'
+          value: sqlAdminPassword
+        }
+        {
+          name: 'appUser'
+          value: appUserPassword 
+        }
+        {
+          name: connectionStringKey
+          value: 'Server=${sqlService.outputs.name}${environment().suffixes.sqlServerHostname}; Database=${!empty(sqlDatabaseName) ? sqlDatabaseName : 'Todo'}; User=${appUser}; Password=${appUserPassword}'
+        }
+      ]
+    }
   }
 }
 
 // The application database
-module sqlServer '../../../../../common/infra/bicep/app/sqlserver.bicep' = {
-  name: 'sql'
+module sqlService 'br/public:avm/res/sql/server:0.2.0' = {
+  name: 'sqlservice'
   scope: rg
   params: {
     name: !empty(sqlServerName) ? sqlServerName : '${abbrs.sqlServers}${resourceToken}'
-    databaseName: sqlDatabaseName
+    administratorLogin: sqlAdmin
+    administratorLoginPassword: sqlAdminPassword
     location: location
     tags: tags
-    sqlAdminPassword: sqlAdminPassword
-    appUserPassword: appUserPassword
-    keyVaultName: keyVault.outputs.name
+    publicNetworkAccess: 'Enabled'
+    databases: [
+      {
+        name: !empty(sqlDatabaseName) ? sqlDatabaseName : 'Todo'
+      }
+    ]
+    firewallRules:[
+      {
+        name: 'Azure Services'
+        startIpAddress: '0.0.0.1'
+        endIpAddress: '255.255.255.254'
+      }
+    ]
+  }
+}
+
+module deploymentScript 'br/public:avm/res/resources/deployment-script:0.1.3' = {
```

Comment:
Can you put this in a common folder?  It can be used by other sql projects in the future.

Reply:
Okay, I've put this in a common folder.
https://github.com/Azure/azure-dev/blob/a35fc18c3ab15d73fc88c49fd0ccb75fc029eb82/templates/todo/common/infra/bicep/app/sql-deployment-script.bicep